### PR TITLE
Fix `addNodeClass` and `addNodeMaterial` in case of renaming classes

### LINKED ISSUE #26518: Nodes module uses class names to register nodes (addNodeClass) which crashes in production
### Description

The `addNodeClass` function exported from `nodes/core/Nodes.js` uses the name property of the nodes classes to register the nodes. The node module works well in development however when the code is packed for production, class names are often modified by the compiler and therefore the whole module crashes.

I believe that this function is mainly used by NodeLoader, but because of this test 
```js
if ( NodeClasses.has( nodeClass.name ) ) throw new Error( `Redefinition of node class ${ nodeClass.name }` );
```
it crashes as soon as any call to this module is made which makes it unusable for production.

### Reproduction steps

1. Import the node module in a file and build for production e.g. webpack

### Code

None

### Live example

None

### Screenshots

_No response_

### Version

r155

### Device

_No response_

### Browser

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by LeviPesin**
Fixes https://github.com/mrdoob/three.js/issues/26518

**Description**

Fix `addNodeClass` and `addNodeMaterial` in case of renaming classes (e.g. when performing minification).

### CODE REVIEW

**Comment by danrossi** [comment]
I checked it out with terser and still get errors

```
webgpu-renderer.module.min.js:6 Uncaught Error: Light  is not a class
```

**Comment by sunag** [comment]
The `addLightNode` should also be included in these changes...

**Comment by LeviPesin** [comment]
Indeed... I will file one more PR to fix it.

**Comment by LeviPesin** [comment]
No, I think there are no errors there...
```js
export function addLightNode( lightClass, lightNodeClass ) {

	if ( LightNodes.has( lightClass ) ) throw new Error( `Redefinition of light node ${ lightNodeClass.name }` );
	if ( typeof lightClass !== 'function' || ! lightClass.name ) throw new Error( `Light ${ lightClass.name } is not a class` );
	if ( typeof lightNodeClass !== 'function' || ! lightNodeClass.name ) throw new Error( `Light node ${ lightNodeClass.name } is not a class` );

	LightNodes.set( lightClass, lightNodeClass );

}
```
Or is it possible that it is created as a class with no name?

File: examples/jsm/nodes/materials/NodeMaterial.js [hunk 1]
```diff
@@ -521,12 +521,13 @@ class NodeMaterial extends ShaderMaterial {
 
 export default NodeMaterial;
 
-export function addNodeMaterial( nodeMaterial ) {
+export function addNodeMaterial( type, nodeMaterial ) {
```
Comment (sunag) [inline examples/jsm/nodes/materials/NodeMaterial.js:524]: I think `nodeMaterialClass` sounds better.

Comment (LeviPesin) [reply examples/jsm/nodes/materials/NodeMaterial.js:524]: Agreed.

Comment (Alexander-103) [reply examples/jsm/nodes/materials/NodeMaterial.js:524]: где можно увидеть звучит лучше nodeMaterialClass?

**Pull request merged**